src/depthai_gstreamer.cpp

```diff
@@ -0,0 +1,932 @@
+      mStreamPlayingCheckTimerId(0),
+      mLoopContext(nullptr)
+{
+    char* argv[] = {""};
+    int argc = 1;
+    gst_init(&argc, reinterpret_cast<char***>(&argv));
+}
+
+void DepthAiGStreamer::Initialize()
+{
+    declare_parameter<std::string>("address", "rtsp://127.0.0.1:8554/mystream");
+    mStreamAddress = this->get_parameter("address").as_string();
+
+    mLoopContext = g_main_context_default();
+    mLoop = g_main_loop_new(mLoopContext, false);
+
+    declare_parameter<std::string>("video_stream_topic", "camera/color/video");
+
+    const std::string video_stream_topic = get_parameter("video_stream_topic").as_string();
+    _video_subscriber = create_subscription<CompressedImageMsg>(video_stream_topic, rclcpp::SystemDefaultsQoS(), std::bind(&DepthAiGStreamer::GrabVideoMsg, this, std::placeholders::_1));
+
+    _video_stream_command_subscriber = this->create_subscription<std_msgs::msg::String>(
+        "videostreamcmd",
+        rclcpp::SystemDefaultsQoS(),
+        std::bind(&DepthAiGStreamer::VideoStreamCommand, this, std::placeholders::_1));
+
+    rcl_interfaces::msg::ParameterDescriptor encoding_desc;
+    encoding_desc.name = "encoding";
+    encoding_desc.type = rclcpp::PARAMETER_STRING;
+    encoding_desc.description = "Encoding format of the video stream.";
+    encoding_desc.additional_constraints = "Accepted values are H264 and H265.";
+    declare_parameter<std::string>("encoding", "H264", encoding_desc);
+    declare_parameter<int>("width", 1280);
+    declare_parameter<int>("height", 720);
+    declare_parameter<int>("fps", 25);
+    declare_parameter<int>("bitrate", 3000000);
+
+    rcl_interfaces::msg::ParameterDescriptor start_stream_on_boot_desc;
+    start_stream_on_boot_desc.name = "start_stream_on_boot";
+    start_stream_on_boot_desc.type = rclcpp::PARAMETER_BOOL;
+    start_stream_on_boot_desc.description =
+        "The node will start the video stream during "
+        "boot if set to true.";
+    start_stream_on_boot_desc.additional_constraints =
+        "This parameter has no "
+        "effect after node has started.";
+
+    declare_parameter<bool>("start_stream_on_boot", true, start_stream_on_boot_desc);
+    rcl_interfaces::msg::ParameterDescriptor address_desc;
+    address_desc.name = "address";
+    address_desc.type = rclcpp::PARAMETER_STRING;
+    address_desc.description =
+        "Video stream destination address. The Gstreamer pipeline is generated "
+        "based on this address.";
+    address_desc.additional_constraints =
+        "UDP or RTSP addresses are accepted. For example:\n"
+        "\trtsp://<user>:<key>@<ip_address>:<port>/<path> (default)\n"
+        "\tudp://<ip_address>:<port>";
+    std::string stream_path =
+        "rtsps://DroneUser:22f6c4de-6144-4f6c-82ea-8afcdf19f316@video-stream.sacplatform.com:8555";
+    std::string ns = std::string(this->get_namespace());
+    declare_parameter<std::string>("address", stream_path + ns, address_desc);
```

Comment:
Address parameter has been declared above in line 75.

Reply:
oops, it was fixed already